include/AudioEngineProfiler.h

```diff
@@ -53,11 +54,31 @@ class AudioEngineProfiler
 
 	void setOutputFile( const QString& outputFile );
 
+	enum class DetailType {
+		NoteSetup,
+		Instruments,
+		Effects,
+		Mixing,
+		DetailCount
+	};
+
+	void startDetail(const DetailType type) { m_detailTimer[static_cast<int>(type)].reset(); }
+	void finishDetail(const DetailType type)
+	{
+		m_detailTime[static_cast<int>(type)] = m_detailTimer[static_cast<int>(type)].elapsed();
+	}
+
+	int detailLoad(const DetailType type) const { return m_detailLoad[static_cast<int>(type)]; }
 
 private:
 	MicroTimer m_periodTimer;
-	int m_cpuLoad;
+	float m_cpuLoad;
 	QFile m_outputFile;
+
+	// Use arrays to avoid dynamic allocations in realtime code
+	std::array<MicroTimer, static_cast<int>(DetailType::DetailCount)> m_detailTimer;
+	std::array<int, static_cast<int>(DetailType::DetailCount)> m_detailTime{0};
+	std::array<float, static_cast<int>(DetailType::DetailCount)> m_detailLoad{0};
```

Comment:
The elements of `m_detailLoad` are written from the audio thread and read from the UI thread, so should use atomics to avoid race conditions. You can use relaxed loads and stores for efficiency if you wish, since there are no data dependencies involved.

Reply:
Good point; I never heard of relaxed loads / stores, so I'll read up on that and make the changes tomorrow-ish.